engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/Atom.java

```diff
@@ -141,6 +143,48 @@ public Object readMember(String member) throws UnknownIdentifierException {
     throw UnknownIdentifierException.create(member);
   }
 
+  @ExportMessage
+  public boolean hasArrayElements() {
+    Map<String, ?> members = constructor.getDefinitionScope().getMethods().get(constructor.getType());
+    return members != null && members.containsKey("at") && members.containsKey("length");
```

Comment:
This operation has to be executable on _fast path_ - e.g. with Truffle partial evaluation, otherwise benchmarks are going to be affected.

However PE mode doesn't work well with `Map` - usually all operations with `Map` have to be behind `@TruffleBoundary` (btw. the `engine-runner-native/buildNativeImage` check should be falling with this change). However `@TruffleBoundary` would be too slow...